litmus-portal/frontend/src/models/analytics.ts

```diff
@@ -1,44 +1,91 @@
-}
-
-export interface Github {
-  stars: string;
-  experimentsCount: string;
-}
-
-export interface GeoCity {
-  name: string;
-  latitude: string;
-  longitude: string;
-  count: string;
-}
-
-export interface SeriesData {
-  date: string;
-  count: string;
-}
-
-export interface Google {
-  totalRuns: string;
-  operatorInstalls: string;
-  geoCity: GeoCity[];
-  geoCountry: string[][];
-  dailyOperatorData: SeriesData[];
-  dailyExperimentData: SeriesData[];
-}
-
-export interface CommunityData {
-  github: Github;
-  google: Google;
-}
-
-interface AnalyticsActionType<T, P> {
-  type: T;
-  payload: P;
-}
-
-export type AnalyticsAction = AnalyticsActionType<
-  typeof AnalyticsActions.LOAD_COMMUNITY_ANALYTICS,
-  CommunityData
->;
+import createReducer from './createReducer';
+import {
+  CommunityData,
+  AnalyticsAction,
+  AnalyticsActions,
+  GeoCity,
+  SeriesData,
+} from '../../models';
+
+const initialState: CommunityData = {
+  github: { stars: '', experimentsCount: '' },
+  google: {
+    totalRuns: '',
+    operatorInstalls: '',
+    geoCity: [],
+    geoCountry: [],
+    dailyExperimentData: [],
+    dailyOperatorData: [],
+    monthlyExperimentData: [],
+    monthlyOperatorData: [],
+  },
+};
+
+export const communityData = createReducer<CommunityData>(initialState, {
+  [AnalyticsActions.LOAD_COMMUNITY_ANALYTICS](
+    state: CommunityData,
+    action: AnalyticsAction
+  ) {
+    const data = action.payload;
+    const geoCity: GeoCity[] = [];
+    data.google.geoCity.forEach((c: any) => {
+      geoCity.push({
+        name: c[0],
+        latitude: c[1],
+        longitude: c[2],
+        count: c[3],
+      });
+    });
+
+    const geoCountry: string[][] = data.google.geoCountry;
+
+    const dailyExperimentData: SeriesData[] = [];
+    data.google.dailyExperimentData.forEach((c: any) => {
+      dailyExperimentData.push({
+        date: c[0],
+        count: c[1],
+      });
+    });
+
+    const dailyOperatorData: SeriesData[] = [];
+    data.google.dailyOperatorData.forEach((c: any) => {
+      dailyOperatorData.push({
+        date: c[0],
+        count: c[1],
+      });
+    });
+
+    const monthlyExperimentData: SeriesData[] = [];
+    data.google.monthlyExperimentData.forEach((c: any) => {
+      monthlyExperimentData.push({
+        date: c[0],
+        count: c[1],
+      });
+    });
+
+    const monthlyOperatorData: SeriesData[] = [];
+    data.google.monthlyOperatorData.forEach((c: any) => {
+      monthlyOperatorData.push({
+        date: c[0],
+        count: c[1],
+      });
+    });
+
+    return {
+      ...state,
+      github: data.github,
+      google: {
+        totalRuns: data.google.totalRuns,
+        operatorInstalls: data.google.operatorInstalls,
+        geoCountry,
+        geoCity,
+        dailyExperimentData,
+        dailyOperatorData,
+        monthlyExperimentData,
+        monthlyOperatorData,
+      },
+    };
+  },
+});
+
+export default communityData;
```

Comment:
This should be in the reducer folder not in the model. You have removed the models file, this won't build. Please run the code and then send PRs. cc: @arkajyotiMukherjee

Reply:
Yeah the bug is now fixed. @gdsoumya and @arkajyotiMukherjee .